File: src/main/java/pl/dmichalski/algorithms/_7_binary_search/Runner.java
package pl.dmichalski.algorithms._7_binary_search;

/**
 * Given a sorted array of integer, write a function
 * called search, that accepts a value and returns the
 * index where the value passed to the function is located.
 * If value is not found, return -1
 */
public class Runner {

    public static void main(String[] args) {
        int[] sortedValues = {1, 2, 3, 4, 5, 6};
        int numberToSearch = 4;

        SearchService searchService = new SearchService();
        BinarySearchService binarySearchService = new BinarySearchService();

        int result1 = searchService.search(sortedValues, numberToSearch);
        int result2 = binarySearchService.search(sortedValues, numberToSearch);

        System.out.println("result1 = " + result1);
        System.out.println("result2 = " + result2);
    }
}

File: src/main/kotlin/pl/dmichalski/algorithms/_7_binary_search/Runner.kt
package pl.dmichalski.algorithms._7_binary_search

/**
 * Given a sorted array of integer, write a function
 * called search, that accepts a value and returns the
 * index where the value passed to the function is located.
 * If value is not found, return -1
 */
object Runner {

    @JvmStatic
    fun main(args: Array<String>) {
        val sortedValues = intArrayOf(1, 2, 3, 4, 5, 6)

        val numberToSearch = 4
        val searchService = SearchService()
        val binarySearchService = BinarySearchService()

        val result1 = searchService.search(sortedValues, numberToSearch)
        val result2 = binarySearchService.search(sortedValues, numberToSearch)

        println("Number $numberToSearch from the array ${sortedValues.contentToString()} is located at the index:")
        println("result1 = $result1")
        println("result2 = $result2")
    }

}
